Migrate wiki articles to Vaadin documentation

eriklumme opened the pull request:
Remake of #9958

<!-- Reviewable:start -->
---
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/vaadin/framework/9959)
<!-- Reviewable:end -->

Commit:
Migrate documentation files and change typo

hesara commented:
Reviewed 5 of 6 files at r1.
Review status: 4 of 5 files reviewed at latest revision, all discussions resolved.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/framework/9959)*
<!-- Sent from Reviewable.io -->

hesara commented:
Reviewed 1 of 6 files at r1.
Review status: all files reviewed at latest revision, all discussions resolved.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/framework/9959)*
<!-- Sent from Reviewable.io -->

Commit:
Migrate OptimizingTheWidgetSet

Commit:
Migrate UsingServerInitiatedEvents

Commit:
Migrate ChooseInputFieldComponentsWisely

Commit:
Migrate CreatingASimpleComponent

Commit:
Migrate IntegratingAnExistingGWTWidget

Commit:
Fix attachment typo

Commit:
Fix typo in the correct place

Commit:
Migrate IntegrationExperiences

hesara commented:
Reviewed 2 of 2 files at r2, 2 of 2 files at r3, 7 of 8 files at r4, 2 of 2 files at r5, 2 of 3 files at r6, 1 of 1 files at r7, 1 of 1 files at r8, 5 of 5 files at r9.
Review status: 19 of 21 files reviewed at latest revision, 5 unresolved discussions.

---

*[documentation/articles/ChooseInputFieldComponentsWisely.asciidoc, line 54 at r4](https://reviewable.io:443/reviews/vaadin/framework/9959#-KtocD2K2L8S3aeoh6XJ:-KtocD2K2L8S3aeoh6XK:b-21l5si) ([raw file](https://github.com/vaadin/framework/blob/bff62f66f949f991a94c890d4ac0592a3ecc7c94/documentation/articles/ChooseInputFieldComponentsWisely.asciidoc#L54)):*
> ```AsciiDoc
> items and small sets of numbers.
> 
> image:img/nativeslect.png[NativeSelect]
> ```

(typo in file name, this link corresponds to the real file name - not a blocker)

---

*[documentation/articles/IntegrationExperiences.asciidoc, line 154 at r9](https://reviewable.io:443/reviews/vaadin/framework/9959#-KtodjfnCO37X30wpYmF:-KtodjfnCO37X30wpYmG:bjwe8n6) ([raw file](https://github.com/vaadin/framework/blob/2cdb0fc351133544879199ea3158a4cef0f0cd13/documentation/articles/IntegrationExperiences.asciidoc#L154)):*
> ```AsciiDoc
> ===============================
> 
> https://new.vaadin.com/directory/component/googlemaps-add-on[GoogleMaps
> ```

this should not have the "new." part in the URL, as it is only temporary/internal

---

*[documentation/articles/IntegrationExperiences.asciidoc, line 279 at r9](https://reviewable.io:443/reviews/vaadin/framework/9959#-Ktodygr7Ev4xy9jzdlC:-Ktodygr7Ev4xy9jzdlD:b-93jum0) ([raw file](https://github.com/vaadin/framework/blob/2cdb0fc351133544879199ea3158a4cef0f0cd13/documentation/articles/IntegrationExperiences.asciidoc#L279)):*
> ```AsciiDoc
> ===============================
> 
> http://www.itmill.com/articles/Packaging_web_applications_for_desktop_use.htm[Packaging
> ```

dead link, not sure if the article is still somewhere

---

*[documentation/articles/OptimizingTheWidgetSet.asciidoc, line 1 at r2](https://reviewable.io:443/reviews/vaadin/framework/9959#-KtobAIQHQV5rmiKPV1C:-KtobAIQHQV5rmiKPV1D:bw42h2t) ([raw file](https://github.com/vaadin/framework/blob/119aad51a2b568619d98156aac4262b70bbf0e18/documentation/articles/OptimizingTheWidgetSet.asciidoc#L1)):*
> ```AsciiDoc
> [[optimizing-the-widget-set]]
> ```

(This functionality is included in later framework versions, though seems to have some issues: #9825 / #9950 . I wonder if we should add some disclaimer that this applies to older versions, though that is the case for a lot of the articles - some of which are for IT Mill Toolkit.)

---

*[documentation/articles/img/vaadingflot.zip, line 0 at r6](https://reviewable.io:443/reviews/vaadin/framework/9959#-KtocljsKfrRrLnXY1C6:-KtocljsKfrRrLnXY1C7:b1kegnf) ([raw file](https://github.com/vaadin/framework/blob/86b49480fdea2d193a5c62cc4644eabe0e9a5a15/documentation/articles/img/vaadingflot.zip#L0)):*
Have you checked that there is nothing suspicious or problematic in the ZIP contents?

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/framework/9959)*
<!-- Sent from Reviewable.io -->

Commit:
Fix image filename typo

Commit:
Fix link typo and remove dead link

hesara commented:
Reviewed 3 of 3 files at r10, 1 of 1 files at r11.
Review status: 19 of 21 files reviewed at latest revision, 1 unresolved discussion.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/framework/9959)*
<!-- Sent from Reviewable.io -->

hesara commented:
Reviewed 1 of 9 files at r4.
Review status: 20 of 21 files reviewed at latest revision, 1 unresolved discussion.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/framework/9959)*
<!-- Sent from Reviewable.io -->

eriklumme commented:
Review status: 20 of 21 files reviewed at latest revision, 4 unresolved discussions.

---

*[documentation/articles/ChooseInputFieldComponentsWisely.asciidoc, line 54 at r4](https://reviewable.io:443/reviews/vaadin/framework/9959#-KtocD2K2L8S3aeoh6XJ:-Ktoh_FFhTgsJBxi1iXM:b6ux3ej) ([raw file](https://github.com/vaadin/framework/blob/bff62f66f949f991a94c890d4ac0592a3ecc7c94/documentation/articles/ChooseInputFieldComponentsWisely.asciidoc#L54)):*
<details><summary><i>Previously, hesara (Henri Sara) wrote…</i></summary><blockquote>

(typo in file name, this link corresponds to the real file name - not a blocker)
</blockquote></details>

File name corrected.

---

*[documentation/articles/IntegrationExperiences.asciidoc, line 154 at r9](https://reviewable.io:443/reviews/vaadin/framework/9959#-KtodjfnCO37X30wpYmF:-KtoirE5XiCZgwzhp-y5:b-896fix) ([raw file](https://github.com/vaadin/framework/blob/2cdb0fc351133544879199ea3158a4cef0f0cd13/documentation/articles/IntegrationExperiences.asciidoc#L154)):*
<details><summary><i>Previously, hesara (Henri Sara) wrote…</i></summary><blockquote>

this should not have the "new." part in the URL, as it is only temporary/internal
</blockquote></details>

Done.

---

*[documentation/articles/IntegrationExperiences.asciidoc, line 279 at r9](https://reviewable.io:443/reviews/vaadin/framework/9959#-Ktodygr7Ev4xy9jzdlC:-KtoitA8hPLsLiQgqDsQ:b-ybn9yt) ([raw file](https://github.com/vaadin/framework/blob/2cdb0fc351133544879199ea3158a4cef0f0cd13/documentation/articles/IntegrationExperiences.asciidoc#L279)):*
<details><summary><i>Previously, hesara (Henri Sara) wrote…</i></summary><blockquote>

dead link, not sure if the article is still somewhere
</blockquote></details>

I could not find the article. I simply removed the link now, as I feel the text makes sense without it.

---

*[documentation/articles/img/vaadingflot.zip, line  at r6](https://reviewable.io:443/reviews/vaadin/framework/9959#-KtocljsKfrRrLnXY1C6:-KtojGoW8k0NFl-cwWqs:b-xzl98z) ([raw file](https://github.com/vaadin/framework/blob/86b49480fdea2d193a5c62cc4644eabe0e9a5a15/documentation/articles/img/vaadingflot.zip#L)):*
<details><summary><i>Previously, hesara (Henri Sara) wrote…</i></summary><blockquote>

Have you checked that there is nothing suspicious or problematic in the ZIP contents?
</blockquote></details>

Yes, I did not find anything suspicious. The /target folder does perhaps not need to be included, but the ZIP file is not particularly large anyway.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/framework/9959)*
<!-- Sent from Reviewable.io -->

Commit:
Migrate VaadinOnGrailsCreateProjectInIntellijIDEA

eriklumme commented:
Review status: 19 of 22 files reviewed at latest revision, 2 unresolved discussions.

---

*[documentation/articles/VaadinOnGrailsCreateProjectInIntelliJIDEA.asciidoc, line 107 at r12](https://reviewable.io:443/reviews/vaadin/framework/9959#-KtoqC07W6PPaPDNqg6a:-KtoqC07W6PPaPDNqg6b:b-igp6ei) ([raw file](https://github.com/vaadin/framework/blob/e16ec959bf6f3c1d76cc7d3ba7a05fb9a0537aa6/documentation/articles/VaadinOnGrailsCreateProjectInIntelliJIDEA.asciidoc#L107)):*
> ```AsciiDoc
> 
> Now you can continue with
> link:VaadinOnGrailsDatabaseAccess.asciidoc[Vaadin
> ```

I changed this to a direct link, but this doc is not yet available. Will add soon.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/framework/9959)*
<!-- Sent from Reviewable.io -->

hesara commented:
Reviewed 1 of 2 files at r12.
Review status: 20 of 22 files reviewed at latest revision, 2 unresolved discussions.

---

*[documentation/articles/img/vaadingflot.zip, line  at r6](https://reviewable.io:443/reviews/vaadin/framework/9959#-KtocljsKfrRrLnXY1C6:-KtorQfWiHtIZr4WjJtv:b-mi5oku) ([raw file](https://github.com/vaadin/framework/blob/86b49480fdea2d193a5c62cc4644eabe0e9a5a15/documentation/articles/img/vaadingflot.zip#L)):*
<details><summary><i>Previously, eriklumme wrote…</i></summary><blockquote>

Yes, I did not find anything suspicious. The /target folder does perhaps not need to be included, but the ZIP file is not particularly large anyway.
</blockquote></details>

I have mixed feelings about this, but I'd perhaps remove target as the ZIP is 15MB that will be in every clone of the framework repository.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/framework/9959)*
<!-- Sent from Reviewable.io -->

eriklumme commented:
*[documentation/articles/img/vaadingflot.zip, line  at r6](https://reviewable.io:443/reviews/vaadin/framework/9959#-KtocljsKfrRrLnXY1C6:-Ktos2lzbpGVeDiJHQRZ:bo64for) ([raw file](https://github.com/vaadin/framework/blob/86b49480fdea2d193a5c62cc4644eabe0e9a5a15/documentation/articles/img/vaadingflot.zip#L)):*
<details><summary><i>Previously, hesara (Henri Sara) wrote…</i></summary><blockquote>

I have mixed feelings about this, but I'd perhaps remove target as the ZIP is 15MB that will be in every clone of the framework repository.
</blockquote></details>

Well that's true. I'll go through it and also compile it myself to see that it works after removing some stuff.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/framework/9959)*
<!-- Sent from Reviewable.io -->

Commit:
Migrate VaadinOnGrailsDatabaseAccess

eriklumme commented:
Review status: 19 of 23 files reviewed at latest revision, 3 unresolved discussions.

---

*[documentation/articles/VaadinOnGrailsCreateProjectInIntelliJIDEA.asciidoc, line 107 at r12](https://reviewable.io:443/reviews/vaadin/framework/9959#-KtoqC07W6PPaPDNqg6a:-Ktow2KbqeNU_fopoY7E:b-896fix) ([raw file](https://github.com/vaadin/framework/blob/e16ec959bf6f3c1d76cc7d3ba7a05fb9a0537aa6/documentation/articles/VaadinOnGrailsCreateProjectInIntelliJIDEA.asciidoc#L107)):*
<details><summary><i>Previously, eriklumme wrote…</i></summary><blockquote>

I changed this to a direct link, but this doc is not yet available. Will add soon.
</blockquote></details>

Done.

---

*[documentation/articles/VaadinOnGrailsDatabaseAccess.asciidoc, line 153 at r13](https://reviewable.io:443/reviews/vaadin/framework/9959#-Ktow4oWjUiKCNqAAJgy:-Ktow4oWjUiKCNqAAJgz:b-3bbqk) ([raw file](https://github.com/vaadin/framework/blob/ae3c4be67668c2fb69f1746d29fad97fbd35903e/documentation/articles/VaadinOnGrailsDatabaseAccess.asciidoc#L153)):*
> ```AsciiDoc
> image:http://vaadinongrails.com/img/idea-datasource.png[IDEA data source]
> 
> You can continue with link:VaadinOnGrailsMultipleUIs.asciidoc[Vaadin on Grails - Multiple UIs]
> ```

Was originally "Vaadin on grails - Localization", but that asciidoc is not included, so I'll just skip that step.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/framework/9959)*
<!-- Sent from Reviewable.io -->

Commit:
Remove unneccessary files

eriklumme commented:
*[documentation/articles/img/vaadingflot.zip, line  at r6](https://reviewable.io:443/reviews/vaadin/framework/9959#-KtocljsKfrRrLnXY1C6:-Ktp0ZhY4zAtxnvIhTOo:b85yk2v) ([raw file](https://github.com/vaadin/framework/blob/86b49480fdea2d193a5c62cc4644eabe0e9a5a15/documentation/articles/img/vaadingflot.zip#L)):*
<details><summary><i>Previously, eriklumme wrote…</i></summary><blockquote>

Well that's true. I'll go through it and also compile it myself to see that it works after removing some stuff.
</blockquote></details>

I removed everything but the source, don't have Eclipse set up with Vaadin but at least I could import it and all files from the article are there. It's 5 kB now.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/framework/9959)*
<!-- Sent from Reviewable.io -->

Commit:
Migrate VaadinOnGrailsMultipleUIs

Commit:
Migrate IntegratingAJavaScriptComponent

Commit:
Change code annotation to JavaScript

hesara commented:
Reviewed 1 of 2 files at r12, 1 of 2 files at r13, 1 of 1 files at r14, 1 of 2 files at r15, 1 of 2 files at r16, 1 of 1 files at r17.
Review status: all files reviewed at latest revision, 1 unresolved discussion.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/framework/9959)*
<!-- Sent from Reviewable.io -->

Commit:
Migrate IntegratingAJavaScriptLibraryAsAnExtension

Commit:
Fix file link

eriklumme commented:
Review status: 24 of 26 files reviewed at latest revision, 2 unresolved discussions.

---

*[documentation/articles/IntegratingAJavaScriptLibraryAsAnExtension.asciidoc, line 19 at r18](https://reviewable.io:443/reviews/vaadin/framework/9959#-Ktp9MfR99R66iUzyJCy:-Ktp9MfR99R66iUzyJCz:b-2w3uen) ([raw file](https://github.com/vaadin/framework/blob/b4d69b981f3f9657abe03523f55c02228f65299c/documentation/articles/IntegratingAJavaScriptLibraryAsAnExtension.asciidoc#L19)):*
> ```AsciiDoc
> function with different arguments, the JavaScript connector logic can be
> equally simple. Aside from asynchronously loading ga.js, the client-side
> code just adds a callback (see link:SimplifiedRPCusingJavaScript.asciidoc[Simplified RPC using JavaScript] for
> ```

This and CreatingAUIExtension have not been migrated yet, so links do not currently work.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/framework/9959)*
<!-- Sent from Reviewable.io -->

Commit:
Migrate CreatingAUIExtension

Commit:
Remove link to non-migrated article

eriklumme commented:
*[documentation/articles/IntegratingAJavaScriptLibraryAsAnExtension.asciidoc, line 19 at r18](https://reviewable.io:443/reviews/vaadin/framework/9959#-Ktp9MfR99R66iUzyJCy:-KtpDYjfL7-2w0bKwU85:b478oq2) ([raw file](https://github.com/vaadin/framework/blob/b4d69b981f3f9657abe03523f55c02228f65299c/documentation/articles/IntegratingAJavaScriptLibraryAsAnExtension.asciidoc#L19)):*
<details><summary><i>Previously, eriklumme wrote…</i></summary><blockquote>

This and CreatingAUIExtension have not been migrated yet, so links do not currently work.
</blockquote></details>

Done. Fixed one link, removed the other as it's Vaadin 7 only and may take time to be migrated (if ever).

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/framework/9959)*
<!-- Sent from Reviewable.io -->

Commit:
Migrate UsingAJavaSscriptLibraryOrAStyleSheetInAnAddOn

Commit:
Migrate ExposingServerSideAPIToJavaScript

Commit:
Migrate UsingRPCFromJavaScript

hesara commented:
Reviewed 1 of 2 files at r20, 1 of 1 files at r21, 2 of 2 files at r22, 2 of 2 files at r23, 2 of 2 files at r24.
Review status: all files reviewed at latest revision, 2 unresolved discussions.

---

*[documentation/articles/CreatingAUIExtension.asciidoc, line 17 at r20](https://reviewable.io:443/reviews/vaadin/framework/9959#-KtpM5YX6mbDd0kCI6RS:-KtpM5YX6mbDd0kCI6RT:bnmxdbp) ([raw file](https://github.com/vaadin/framework/blob/0c429f7739771c0abab7326f906e4a3e05603c88/documentation/articles/CreatingAUIExtension.asciidoc#L17)):*
> ```AsciiDoc
> are not part of the regular layout hierarchy.
> 
> We will rewrite the
> ```

(this example as well is quite redundant with UI.setPollingInterval(), but I guess still ok as example)

---

*[documentation/articles/IntegratingAJavaScriptLibraryAsAnExtension.asciidoc, line 19 at r18](https://reviewable.io:443/reviews/vaadin/framework/9959#-Ktp9MfR99R66iUzyJCy:-KtpLeT4dTBEmINMOjHd:b-p0ypmc) ([raw file](https://github.com/vaadin/framework/blob/b4d69b981f3f9657abe03523f55c02228f65299c/documentation/articles/IntegratingAJavaScriptLibraryAsAnExtension.asciidoc#L19)):*
<details><summary><i>Previously, eriklumme wrote…</i></summary><blockquote>

Done. Fixed one link, removed the other as it's Vaadin 7 only and may take time to be migrated (if ever).
</blockquote></details>

Why is it Vaadin 7 only? (just wondering, as many of the articles do concern mostly very old versions)

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/framework/9959)*
<!-- Sent from Reviewable.io -->

eriklumme commented:
*[documentation/articles/IntegratingAJavaScriptLibraryAsAnExtension.asciidoc, line 19 at r18](https://reviewable.io:443/reviews/vaadin/framework/9959#-Ktp9MfR99R66iUzyJCy:-KtpO3If6zJHyhJgfa_D:b-293p1j) ([raw file](https://github.com/vaadin/framework/blob/b4d69b981f3f9657abe03523f55c02228f65299c/documentation/articles/IntegratingAJavaScriptLibraryAsAnExtension.asciidoc#L19)):*
<details><summary><i>Previously, hesara (Henri Sara) wrote…</i></summary><blockquote>

Why is it Vaadin 7 only? (just wondering, as many of the articles do concern mostly very old versions)
</blockquote></details>

In https://docs.google.com/spreadsheets/d/1vS1l76YEBWPzAwyMG1k3PaV42ZlmAiuL14HaDsuZOMg/ the Vaadin 7&8 will be migrated first. "Simplified RPC using JavaScript" is listed as "7 only". I'm not sure if/when it will be migrated.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/framework/9959)*
<!-- Sent from Reviewable.io -->

Commit:
Migrate IBGettingStartedWithVaadinSpringWithoutSPringBoot

Commit:
Migrate Vaadin7SpringSecurityBaseAuthentication

Commit:
Migrate UsingBeanValidationToValidateInput

Commit:
Migrate VaadinSpringTips

Commit:
Add formatting for properties

Commit:
Migrate VaadinCDI

Commit:
Migrate IIInjectionAndScopes

hesara commented:
Reviewed 3 of 3 files at r25, 2 of 2 files at r26, 2 of 2 files at r27, 2 of 2 files at r28, 1 of 1 files at r29, 2 of 2 files at r30, 5 of 5 files at r31.
Review status: all files reviewed at latest revision, 5 unresolved discussions.

---

*[documentation/articles/IBGettingStartedWithVaadinSpringWithoutSpringBoot.asciidoc, line 11 at r25](https://reviewable.io:443/reviews/vaadin/framework/9959#-KtpczdcST4lmkYSIEhB:-KtpczdcST4lmkYSIEhC:bljt3t0) ([raw file](https://github.com/vaadin/framework/blob/f8938797b75572d9a1809c747dda45482eb1bb68/documentation/articles/IBGettingStartedWithVaadinSpringWithoutSpringBoot.asciidoc#L11)):*
> ```AsciiDoc
> application to a server. 
> 
> *Note that this tutorial is for using Vaadin Spring without Spring Boot.
> ```

(note that https://vaadin.github.io/spring-tutorial/#getting-started-no-boot also covers this case, though not quite like this tutorial)

---

*[documentation/articles/IBGettingStartedWithVaadinSpringWithoutSpringBoot.asciidoc, line 17 at r25](https://reviewable.io:443/reviews/vaadin/framework/9959#-KtpdDELwUu19CJou3_E:-KtpdDELwUu19CJou3_F:b-bkjzan) ([raw file](https://github.com/vaadin/framework/blob/f8938797b75572d9a1809c747dda45482eb1bb68/documentation/articles/IBGettingStartedWithVaadinSpringWithoutSpringBoot.asciidoc#L17)):*
> ```AsciiDoc
> 
> *For more information on using Vaadin Spring without Spring Boot, see
> https://vaadin.com/wiki/-/wiki/Spring+Vaadin/IV+-+Configuring+and+Using+Vaadin+Spring+without+Spring+Boot[this
> ```

several old wiki links in the file

---

*[documentation/articles/IIInjectionAndScopes.asciidoc, line 1 at r31](https://reviewable.io:443/reviews/vaadin/framework/9959#-KtpeJ93T3BPYuiM6dAx:-KtpeJ93T3BPYuiM6dAy:btssdo4) ([raw file](https://github.com/vaadin/framework/blob/8e541d6aca977b11eafe0463d132a6171600d2e9/documentation/articles/IIInjectionAndScopes.asciidoc#L1)):*
> ```AsciiDoc
> [[ii-injection-and-scopes]]
> ```

should have a link to this from the CDI tutorial master page

---

*[documentation/articles/UsingBeanValidationToValidateInput.asciidoc, line 1 at r27](https://reviewable.io:443/reviews/vaadin/framework/9959#-KtpdaQR72T76lMkvPsm:-KtpdaQR72T76lMkvPsn:bh5jxbm) ([raw file](https://github.com/vaadin/framework/blob/0f531d314546ba896cd75a7765709e6aa2841215/documentation/articles/UsingBeanValidationToValidateInput.asciidoc#L1)):*
> ```AsciiDoc
> [[using-bean-validation-to-validate-input]]
> ```

(this as well is mostly for V7 and old data binding APIs, but I guess can be included here)

---

*[documentation/articles/VaadinCDI.asciidoc, line 1 at r30](https://reviewable.io:443/reviews/vaadin/framework/9959#-Ktpe4q6pffNnxBHJxQ4:-Ktpe4q6pffNnxBHJxQ5:b-suw7i3) ([raw file](https://github.com/vaadin/framework/blob/ef1dbdcd7f9cdf914edff324c759e00054e83136/documentation/articles/VaadinCDI.asciidoc#L1)):*
> ```AsciiDoc
> [[vaadin-cdi]]
> ```

(just a master page for the tutorials, should probably have links once the parts are migrated)

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/framework/9959)*
<!-- Sent from Reviewable.io -->

Commit:
Migrate CreatingSecureVaadinApplicationsUsingJEE6

eriklumme commented:
Review status: 39 of 47 files reviewed at latest revision, 5 unresolved discussions.

---

*[documentation/articles/CreatingSecureVaadinApplicationsUsingJEE6.asciidoc, line 55 at r32](https://reviewable.io:443/reviews/vaadin/framework/9959#-KtpjTe87oSjZBkBXksJ:-KtpjTe87oSjZBkBXksK:bcc0cgd) ([raw file](https://github.com/vaadin/framework/blob/0c4a4ee6410a69d97036a90e6562981adabb6e1e/documentation/articles/CreatingSecureVaadinApplicationsUsingJEE6.asciidoc#L55)):*
> ```AsciiDoc
> 
> The source code can be downloaded
> https://vaadin.com/wiki?p_auth=jN6Filxv&p_p_id=36&p_p_lifecycle=1&p_p_state=exclusive&p_p_mode=view&p_p_col_id=row-1&p_p_col_pos=1&p_p_col_count=2&_36_struts_action=%2Fwiki%2Fget_page_attachment&p_r_p_185834411_nodeId=10674&p_r_p_185834411_nodeId=10674&p_r_p_185834411_nodeId=10674&p_r_p_185834411_title=Creating+Secure+Vaadin+Applications+using+JEE6&p_r_p_185834411_title=Creating+Secure+Vaadin+Applications+using+JEE6&_36_fileName=SecureVaadinApplicationDemo.zip[here]
> ```

Link is broken. I have the attachment and can attach it, but it's an almost 15 MB ZIP-file again that includes a couple JARs as well.

---

*[documentation/articles/VaadinCDI.asciidoc, line 1 at r30](https://reviewable.io:443/reviews/vaadin/framework/9959#-Ktpe4q6pffNnxBHJxQ4:-KtpjgBY6elogffcV4vO:b77z027) ([raw file](https://github.com/vaadin/framework/blob/ef1dbdcd7f9cdf914edff324c759e00054e83136/documentation/articles/VaadinCDI.asciidoc#L1)):*
<details><summary><i>Previously, hesara (Henri Sara) wrote…</i></summary><blockquote>

(just a master page for the tutorials, should probably have links once the parts are migrated)
</blockquote></details>

Alright. It seems to me that all parts aren't included in the files I have, though.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/framework/9959)*
<!-- Sent from Reviewable.io -->

Commit:
Add CDI Injection link from master doc

Commit:
Migrate UsingVaadinCDIWithJAASAuthentication

Commit:
Add missing file changes

Commit:
Migrate LoadTestingWithGatling

Commit:
Replace wiki links with vaadin.github.io/spring-tutorial

eriklumme commented:
Review status: 37 of 49 files reviewed at latest revision, 5 unresolved discussions.

---

*[documentation/articles/IBGettingStartedWithVaadinSpringWithoutSpringBoot.asciidoc, line 11 at r25](https://reviewable.io:443/reviews/vaadin/framework/9959#-KtpczdcST4lmkYSIEhB:-Ktq6R75a_uBAEZOP5aj:b9oj1fs) ([raw file](https://github.com/vaadin/framework/blob/f8938797b75572d9a1809c747dda45482eb1bb68/documentation/articles/IBGettingStartedWithVaadinSpringWithoutSpringBoot.asciidoc#L11)):*
<details><summary><i>Previously, hesara (Henri Sara) wrote…</i></summary><blockquote>

(note that https://vaadin.github.io/spring-tutorial/#getting-started-no-boot also covers this case, though not quite like this tutorial)
</blockquote></details>

Text changed to reflect that more info is found in the tutorial.

---

*[documentation/articles/IBGettingStartedWithVaadinSpringWithoutSpringBoot.asciidoc, line 17 at r25](https://reviewable.io:443/reviews/vaadin/framework/9959#-KtpdDELwUu19CJou3_E:-Ktq6Zz9_rN27XaEzdUn:b-gq6meb) ([raw file](https://github.com/vaadin/framework/blob/f8938797b75572d9a1809c747dda45482eb1bb68/documentation/articles/IBGettingStartedWithVaadinSpringWithoutSpringBoot.asciidoc#L17)):*
<details><summary><i>Previously, hesara (Henri Sara) wrote…</i></summary><blockquote>

several old wiki links in the file
</blockquote></details>

Links removed/change to github.io tutorial

---

*[documentation/articles/IIInjectionAndScopes.asciidoc, line 1 at r31](https://reviewable.io:443/reviews/vaadin/framework/9959#-KtpeJ93T3BPYuiM6dAx:-KtpqSyTWg3nJxxVqoc7:b-896fix) ([raw file](https://github.com/vaadin/framework/blob/8e541d6aca977b11eafe0463d132a6171600d2e9/documentation/articles/IIInjectionAndScopes.asciidoc#L1)):*
<details><summary><i>Previously, hesara (Henri Sara) wrote…</i></summary><blockquote>

should have a link to this from the CDI tutorial master page
</blockquote></details>

Done.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/framework/9959)*
<!-- Sent from Reviewable.io -->

Commit:
Migrate VaadinScalabilityTestingWIthAmazonWebServices

Commit:
Add missing changes

Commit:
Fix formatting error

Commit:
Migrate UsingFontIcons

Commit:
Migrate DynamicallyInjectingCSS

Commit:
Migrate ValoExamples

Commit:
Migrate ReadOnlyVsDisabledFields

Commit:
Migrate CreatingAThemeUsingSass

Commit:
Migrate ValoThemeGettingStarted

Commit:
Migrate UseTooltipsToClarifyFunctions

Commit:
Migrate EnableAndDisableButtonsToIndicateState

Commit:
Migrate ChangingThemeOnTheFly

Commit:
Migrate MarkRequiredFieldsAsSuch

Commit:
Migrate PackagingSCSSorCSSinAnAddOn

Commit:
Migrate RightAlignComparableNumericalFields

Commit:
Migrate CustomizingComponentThemeWithSass

Commit:
Fix indentation

hesara commented:
Reviewed 8 of 8 files at r32.
Review status: 44 of 79 files reviewed at latest revision, 5 unresolved discussions.

---

*[documentation/articles/CreatingSecureVaadinApplicationsUsingJEE6.asciidoc, line 55 at r32](https://reviewable.io:443/reviews/vaadin/framework/9959#-KtpjTe87oSjZBkBXksJ:-KtuJLzS0p4hzPieikUM:bwx5juz) ([raw file](https://github.com/vaadin/framework/blob/0c4a4ee6410a69d97036a90e6562981adabb6e1e/documentation/articles/CreatingSecureVaadinApplicationsUsingJEE6.asciidoc#L55)):*
<details><summary><i>Previously, eriklumme wrote…</i></summary><blockquote>

Link is broken. I have the attachment and can attach it, but it's an almost 15 MB ZIP-file again that includes a couple JARs as well.
</blockquote></details>

Is it a Maven project? Perhaps we could strip out things that can be found in Maven Central, and just list them if not a Maven or Ivy project.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/framework/9959)*
<!-- Sent from Reviewable.io -->

eriklumme commented:
*[documentation/articles/CreatingSecureVaadinApplicationsUsingJEE6.asciidoc, line 55 at r32](https://reviewable.io:443/reviews/vaadin/framework/9959#-KtpjTe87oSjZBkBXksJ:-KtuKD_psJkc4D0eRw5J:b-4d2te8) ([raw file](https://github.com/vaadin/framework/blob/0c4a4ee6410a69d97036a90e6562981adabb6e1e/documentation/articles/CreatingSecureVaadinApplicationsUsingJEE6.asciidoc#L55)):*
<details><summary><i>Previously, hesara (Henri Sara) wrote…</i></summary><blockquote>

Is it a Maven project? Perhaps we could strip out things that can be found in Maven Central, and just list them if not a Maven or Ivy project.
</blockquote></details>

Seems to be an Ant project packaged for use in Netbeans.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/framework/9959)*
<!-- Sent from Reviewable.io -->

hesara commented:
Reviewed 5 of 35 files at r33.
Review status: 49 of 79 files reviewed at latest revision, 6 unresolved discussions.

---

*[a discussion](https://reviewable.io:443/reviews/vaadin/framework/9959#-KtuKiYp30PKStwN-BDt:-KtuKiYp30PKStwN-BDu:b-mf9q39) (no related file):*
These are starting to get so bulky and so mixed with respect to applicable versions, quality etc., that I'm more and more questioning the decision to include these in the framework repository. I'd suggest extracting the old and new ones to a separate repository instead, and somehow updating documentation builds to use it.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/framework/9959)*
<!-- Sent from Reviewable.io -->

Commit:
Migrate WidgetStylingUsingOnlyCSS

Commit:
Migrate VisuallyDistinguishPrimaryActions

Commit:
Migrate LabelButtonsExpressively

Commit:
Migrate CreatingAServlet3.0Application

Commit:
Migrate CreatingANEclipseProject

Commit:
Migrate CreatingASimpleComponentContainer

Commit:
Migrate UsingRPCToSendEventsToTheClient

Commit:
Migrate CreatingAComponentExtension

Commit:
Migrate UsingDeclarativeServices

Commit:
Migrate DynamicallyUpdatingStateBeforeSendingChangesToClient

Commit:
Migrate GettingStartedOnNetBeans

Commit:
Migrate ComponentAddonProjectSetupHowto

Commit:
Migrate OpeningAUIInAPopupWindow

Commit:
Migrate ViewChangeConfirmations

Commit:
Migrate CreatingABookmarkableApplicationWithBackButtonSupport

Commit:
Migrate BroadcastingMessagesToOtherUsers

Commit:
Migrate ConfigureInputFieldsToGuideDataEntry

Commit:
Migrate AddingASplashScreen and CreatingMultiTabApplications

eriklumme commented:
@hezamu Any input on Sara's comment above? About moving these to separate repository.

pleku commented:
@hesara what gains would it bring to separate these to yet another repository, to another place from the main product documentation ? Is there something with the current approach that makes it cumbersome ? Is there some technical difficulties ? These all should still apply for FW version 8, that has been verified already.

These wiki articles are for the community by the community and we take no responsibility on the content. Thus the review should be mainly about verifying that the content is not total non-sense and it is in the right location.

Commit:
Fix typ and add attachment link

eriklumme commented:
Review status: 47 of 115 files reviewed at latest revision, 6 unresolved discussions.

---

*[documentation/articles/CreatingSecureVaadinApplicationsUsingJEE6.asciidoc, line 55 at r32](https://reviewable.io:443/reviews/vaadin/framework/9959#-KtpjTe87oSjZBkBXksJ:-Ku4QF7LF1AlRNS002yJ:bhi7ezm) ([raw file](https://github.com/vaadin/framework/blob/0c4a4ee6410a69d97036a90e6562981adabb6e1e/documentation/articles/CreatingSecureVaadinApplicationsUsingJEE6.asciidoc#L55)):*
<details><summary><i>Previously, eriklumme wrote…</i></summary><blockquote>

Seems to be an Ant project packaged for use in Netbeans.
</blockquote></details>

I made a new repo on my account for the attachment and linked to there.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/framework/9959)*
<!-- Sent from Reviewable.io -->

hesara commented:
Reviewed 30 of 35 files at r33, 2 of 2 files at r34, 6 of 6 files at r35, 4 of 5 files at r36, 2 of 2 files at r37, 5 of 5 files at r38, 2 of 2 files at r39, 2 of 2 files at r40, 2 of 2 files at r41, 2 of 2 files at r42, 2 of 2 files at r43, 4 of 4 files at r44, 2 of 2 files at r45, 2 of 2 files at r46, 2 of 2 files at r47, 2 of 2 files at r48, 2 of 2 files at r49, 7 of 7 files at r50, 3 of 3 files at r51, 1 of 1 files at r52.
Review status: all files reviewed at latest revision, 5 unresolved discussions.

---

*[a discussion](https://reviewable.io:443/reviews/vaadin/framework/9959#-KtuKiYp30PKStwN-BDt:-KuImzCqzx4WhvryqARJ:b-1v0dhe) (no related file):*
<details><summary><i>Previously, hesara (Henri Sara) wrote…</i></summary><blockquote>

These are starting to get so bulky and so mixed with respect to applicable versions, quality etc., that I'm more and more questioning the decision to include these in the framework repository. I'd suggest extracting the old and new ones to a separate repository instead, and somehow updating documentation builds to use it.
</blockquote></details>

Based on comments by others, let's go ahead with this to get at least one version of each article saved, and then in a separate operation (after merging this) clean up some obsolete articles etc. If there are very large attachments, it is nicer to have them outside the framework repository.

---

*[documentation/articles/UsingDeclarativeServices.asciidoc, line 1 at r42](https://reviewable.io:443/reviews/vaadin/framework/9959#-KuIlePpXwFdxiqisW-K:-KuIlePpXwFdxiqisW-L:b-bu0cme) ([raw file](https://github.com/vaadin/framework/blob/fbd7aca75107abc347f27d014ba176d85a2f9991/documentation/articles/UsingDeclarativeServices.asciidoc#L1)):*
> ```AsciiDoc
> [[using-declarative-services]]
> ```

(some things may have changed in 8.1, but keep this anyway and let someone update later if necessary)

---

*[documentation/articles/UsingDeclarativeServices.asciidoc, line 16 at r42](https://reviewable.io:443/reviews/vaadin/framework/9959#-KuIlUNn_Z493_2wO3-0:-KuIlUNo6R1Ww_08sa0i:b-17nmyq) ([raw file](https://github.com/vaadin/framework/blob/fbd7aca75107abc347f27d014ba176d85a2f9991/documentation/articles/UsingDeclarativeServices.asciidoc#L16)):*
> ```AsciiDoc
> OSGi-DS is enough to understand the example defined below.
> 
> [Error: Macro 'TableOfContents' doesn't exist]
> ```

Error message in migrated document? 

---

*[documentation/articles/VaadinCDI.asciidoc, line 1 at r30](https://reviewable.io:443/reviews/vaadin/framework/9959#-Ktpe4q6pffNnxBHJxQ4:-KuIwaiyUR2SvHyvi-Zk:b-rflgnl) ([raw file](https://github.com/vaadin/framework/blob/ef1dbdcd7f9cdf914edff324c759e00054e83136/documentation/articles/VaadinCDI.asciidoc#L1)):*
<details><summary><i>Previously, eriklumme wrote…</i></summary><blockquote>

Alright. It seems to me that all parts aren't included in the files I have, though.
</blockquote></details>

links from here to all the migrated CDI tutorial parts?

---

*[documentation/articles/VaadinScalabilityTestingWithAmazonWebServices.asciidoc, line 21 at r52](https://reviewable.io:443/reviews/vaadin/framework/9959#-KuIugMMoAj1bk3b3Ktb:-KuIugMMoAj1bk3b3Ktc:bodzgh6) ([raw file](https://github.com/vaadin/framework/blob/f08a6a8ac783e6800659a8c77b592e7b35d16419/documentation/articles/VaadinScalabilityTestingWithAmazonWebServices.asciidoc#L21)):*
> ```AsciiDoc
> http://jakarta.apache.org/jmeter/[Apache JMeter], MySQL and Linux shell
> usage. You will also need to know how to checkout the
> http://dev.vaadin.com/svn/incubator/QuickTickets/trunk/[QuickTickets
> ```

(this SVN server might not be around forever, but is still at the moment - at some point we probably need to migrate the application to GitHub)

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/framework/9959)*
<!-- Sent from Reviewable.io -->

Commit:
Remove migration error message

hesara commented:
Reviewed 1 of 1 files at r53.
Review status: all files reviewed at latest revision, 1 unresolved discussion.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/framework/9959)*
<!-- Sent from Reviewable.io -->

eriklumme commented:
Review status: all files reviewed at latest revision, 2 unresolved discussions.

---

*[documentation/articles/UsingDeclarativeServices.asciidoc, line 16 at r42](https://reviewable.io:443/reviews/vaadin/framework/9959#-KuIlUNn_Z493_2wO3-0:-KuIyeCwbLBtYueF45lt:b-v18ldz) ([raw file](https://github.com/vaadin/framework/blob/fbd7aca75107abc347f27d014ba176d85a2f9991/documentation/articles/UsingDeclarativeServices.asciidoc#L16)):*
<details><summary><i>Previously, hesara (Henri Sara) wrote…</i></summary><blockquote>

Error message in migrated document? 
</blockquote></details>

Fixed. Did a quick search and seems that was the only one I missed.

---

*[documentation/articles/VaadinCDI.asciidoc, line 1 at r30](https://reviewable.io:443/reviews/vaadin/framework/9959#-Ktpe4q6pffNnxBHJxQ4:-KuJ-hZv6jbaKU_yEzMc:b-uumfji) ([raw file](https://github.com/vaadin/framework/blob/ef1dbdcd7f9cdf914edff324c759e00054e83136/documentation/articles/VaadinCDI.asciidoc#L1)):*
<details><summary><i>Previously, hesara (Henri Sara) wrote…</i></summary><blockquote>

links from here to all the migrated CDI tutorial parts?
</blockquote></details>

The only other part I can find is V - Access control. However, it is "7 only" and thus not included in this branch, so a standard `link:<file>` wouldn't work. The other parts do not seem to be included in the articles directory at all.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/framework/9959)*
<!-- Sent from Reviewable.io -->

hesara commented:
Review status: all files reviewed at latest revision, 1 unresolved discussion.

---

*[documentation/articles/VaadinCDI.asciidoc, line 1 at r30](https://reviewable.io:443/reviews/vaadin/framework/9959#-Ktpe4q6pffNnxBHJxQ4:-KuJ3bWXrrMbJj-diU7e:b-sky63z) ([raw file](https://github.com/vaadin/framework/blob/ef1dbdcd7f9cdf914edff324c759e00054e83136/documentation/articles/VaadinCDI.asciidoc#L1)):*
<details><summary><i>Previously, eriklumme wrote…</i></summary><blockquote>

The only other part I can find is V - Access control. However, it is "7 only" and thus not included in this branch, so a standard `link:<file>` wouldn't work. The other parts do not seem to be included in the articles directory at all.
</blockquote></details>

Ok, the others like the JAAS article seem to be outside the series.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/framework/9959)*
<!-- Sent from Reviewable.io -->

hesara commented:
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>

---

Review status: all files reviewed at latest revision, all discussions resolved, some commit checks pending.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/framework/9959#-:-KuJ3lzkYVRgUMsn-hzw:bnfp4nl)*
<!-- Sent from Reviewable.io -->
